kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/model/ingress/Ingresses.java

```diff
@@ -12,11 +12,12 @@
 
 import io.kroxylicious.kubernetes.api.v1alpha1.KafkaProxy;
 import io.kroxylicious.kubernetes.api.v1alpha1.KafkaProxyIngress;
+import io.kroxylicious.kubernetes.api.v1alpha1.KafkaService;
 import io.kroxylicious.kubernetes.api.v1alpha1.VirtualKafkaCluster;
 import io.kroxylicious.kubernetes.api.v1alpha1.kafkaproxyingressspec.ClusterIP;
 import io.kroxylicious.kubernetes.api.v1alpha1.kafkaservicespec.NodeIdRanges;
 import io.kroxylicious.kubernetes.api.v1alpha1.kafkaservicespec.NodeIdRangesBuilder;
-import io.kroxylicious.kubernetes.operator.resolver.ProxyResolutionResult;
+import io.kroxylicious.kubernetes.operator.resolver.ClusterResolutionResult;
 
 class Ingresses {
```

Comment:
Please can we have Javadoc that explains what an `Ingresses` is, conceptually speaking?

We have:
* IngressDefinition
* IngressInstance
* Ingresses
* ProxyIngressModel

I have a real hard time understanding what's going on there.

Reply:
I don't want to go too far down tidying the ingress of things up as it's a mess. I'll rename `Ingresses` to `IngressDefinitionBuilder` in this PR to give it a better name and disambiguate from the `Ingresses` class generated (with the annoying pluralization) from the CR.

Reply:
ah sorry, you were requesting javadoc, I'll have a shot at improving it

Reply:
Thanks! I hadn't even clocked that we had two `Ingresses`. Definitely distinct names would be good.